File: src/IO_24/_02_Бондаренко_Тарас_Андрійович/lab5/Letter.java
package IO_24._02_Бондаренко_Тарас_Андрійович.lab5;

public class Letter {
    private final char character;

    public Letter(char character) {
        this.character = character;
    }

    @Override
    public String toString() {
        return String.valueOf(character);
    }
}

File: Java/lab_5/Letter.kt
package OOP.Java.lab_5

/**
 * A class representing a single letter in a word.
 *
 * @property character a character value of a Letter.
 */
class Letter(private val character: Char) {
    override fun toString(): String {
        return this.character.toString()
    }

    /**
     * Indicates whether the two objects of class Letter are "equal".
     *
     * @param letter a letter to compare.
     * @param ignoreCase if set to "true" will ignore the case of Letters compared.
     * @return "true" if letters are equal and "false" if they aren't.
     */
    fun letterEquals(letter: Letter, ignoreCase: Boolean): Boolean {
        return this.character.toString().equals(letter.toString(), ignoreCase)
    }
}